fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/service/TwoFactorConfigurationServiceImpl.java

```diff
@@ -0,0 +1,293 @@
+            TwoFactorConfiguration configuration = configurationRepository.findByName(parameterName);
+            if(configuration == null) {
+                continue;
+            }
+
+            if(command.isChangeInIntegerSansLocaleParameterNamed(parameterName, configuration.getIntegerValue())) {
+                final Long newValue = command.longValueOfParameterNamed(parameterName);
+                actualChanges.put(parameterName, newValue);
+                configuration.setIntegerValue(newValue);
+                configurationRepository.save(configuration);
+            }
+        }
+
+        if(!actualChanges.isEmpty()) {
+            configurationRepository.flush();
+        }
+
+        return actualChanges;
+    }
+
+    public boolean isSMSEnabled() {
+        return getBooleanConfig(TwoFactorConfigurationConstants.ENABLE_SMS_DELIVERY, false);
+    }
+
+    @Override
+    public Integer getSMSProviderId() {
+        Integer value = getIntegerConfig(TwoFactorConfigurationConstants.SMS_PROVIDER_ID,
+                null);
+        if(value < 1) {
+            return null;
+        }
+        return value;
+    }
+
+    @Override
+    public String getSmsText() {
+        return getStringConfig(TwoFactorConfigurationConstants.SMS_MESSAGE_TEXT, DEFAULT_SMS_TEXT);
+    }
+
+    @Override
+    public boolean isEmailEnabled() {
+        return getBooleanConfig(TwoFactorConfigurationConstants.ENABLE_EMAIL_DELIVERY, false);
+    }
+
+    @Override
+    public String getEmailSubject() {
+        return getStringConfig(TwoFactorConfigurationConstants.EMAIL_SUBJECT, DEFAULT_EMAIL_SUBJECT);
+    }
+
+    @Override
+    public String getEmailBody() {
+        return getStringConfig(TwoFactorConfigurationConstants.EMAIL_BODY, DEFAULT_EMAIL_BODY);
+    }
+
+    @Override
+    public String getFormattedEmailSubjectFor(AppUser user, OTPRequest request) {
+        final Map<String, Object> templateData = processTemplateDataFor(user, request);
+        return compileTextTemplate(getEmailSubject(), TwoFactorConstants.EMAIL_DELIVERY_METHOD_NAME, templateData);
+    }
+
+    @Override
+    public String getFormattedEmailBodyFor(AppUser user, OTPRequest request) {
+        final Map<String, Object> templateData = processTemplateDataFor(user, request);
+        return compileTextTemplate(getEmailBody(), TwoFactorConstants.EMAIL_DELIVERY_METHOD_NAME, templateData);
+    }
+
+    @Override
+    public String getFormattedSmsTextFor(AppUser user, OTPRequest request) {
+        final Map<String, Object> templateData = processTemplateDataFor(user, request);
+        return compileTextTemplate(getSmsText(), TwoFactorConstants.SMS_DELIVERY_METHOD_NAME, templateData);
+    }
+
+    @Override
+    public Integer getOTPTokenLength() {
+        Integer defaultValue = 5;
+        Integer value = getIntegerConfig(TwoFactorConfigurationConstants.OTP_TOKEN_LENGTH,
+                defaultValue);
+        if(value < 1) {
```

Comment:
instead of checking here for value less than 1 , from getInteger method only return default value